data/demo/data/valid-states.csv

```diff
@@ -1,21 +1,20 @@
-setting,setting-group,min-academic-year,max-academic-year,needs,setting->setting
-EO,Other,-4,20,"CI,CL,OTH,SEMH,SP,UKN","EO,EYE,FEC,IMS,ISS,ISSR,MAP,MMS,MMSIB,MMSOB,MSS,MSSIB,MSSOB,MSSR,MU,MUOB,NMSS,NMSSR,OOE"
-EYS,Mainstream,-4,0,"CI,CL,OTH,SEMH,SP,UKN","EO,EYE,FEC,IMS,ISS,ISSR,MAP,MMS,MMSIB,MMSOB,MSS,MSSIB,MSSOB,MSSR,MU,MUOB,NMSS,NMSSR,OOE"
-FEC,Further education,12,20,"CI,CL,OTH,SEMH,SP,UKN","EO,EYE,FEC,IMS,ISS,ISSR,MAP,MMS,MMSIB,MMSOB,MSS,MSSIB,MSSOB,MSSR,MU,MUOB,NMSS,NMSSR,OOE"
-IMS,Mainstream,-2,14,"CI,CL,OTH,SEMH,SP,UKN","EO,EYE,FEC,IMS,ISS,ISSR,MAP,MMS,MMSIB,MMSOB,MSS,MSSIB,MSSOB,MSSR,MU,MUOB,NMSS,NMSSR,OOE"
-ISS,Special,-4,15,"CI,CL,OTH,SEMH,SP,UKN","EO,EYE,FEC,IMS,ISS,ISSR,MAP,MMS,MMSIB,MMSOB,MSS,MSSIB,MSSOB,MSSR,MU,MUOB,NMSS,NMSSR,OOE"
-ISSR,Special,-4,15,"CI,CL,OTH,SEMH,SP,UKN","EO,EYE,FEC,IMS,ISS,ISSR,MAP,MMS,MMSIB,MMSOB,MSS,MSSIB,MSSOB,MSSR,MU,MUOB,NMSS,NMSSR,OOE"
-MAP,Other,7,20,"CI,CL,OTH,SEMH,SP,UKN","EO,EYE,FEC,IMS,ISS,ISSR,MAP,MMS,MMSIB,MMSOB,MSS,MSSIB,MSSOB,MSSR,MU,MUOB,NMSS,NMSSR,OOE"
-MMS,Mainstream,-2,14,"CI,CL,OTH,SEMH,SP,UKN","EO,EYE,FEC,IMS,ISS,ISSR,MAP,MMS,MMSIB,MMSOB,MSS,MSSIB,MSSOB,MSSR,MU,MUOB,NMSS,NMSSR,OOE"
-MMSIB,Special,-2,14,"CI,CL,OTH,SEMH,SP,UKN","EO,EYE,FEC,IMS,ISS,ISSR,MAP,MMS,MMSIB,MMSOB,MSS,MSSIB,MSSOB,MSSR,MU,MUOB,NMSS,NMSSR,OOE"
-MMSOB,Mainstream,-2,14,"CI,CL,OTH,SEMH,SP,UKN","EO,EYE,FEC,IMS,ISS,ISSR,MAP,MMS,MMSIB,MMSOB,MSS,MSSIB,MSSOB,MSSR,MU,MUOB,NMSS,NMSSR,OOE"
-MSS,Special,-4,20,"CI,CL,OTH,SEMH,SP,UKN","EO,EYE,FEC,IMS,ISS,ISSR,MAP,MMS,MMSIB,MMSOB,MSS,MSSIB,MSSOB,MSSR,MU,MUOB,NMSS,NMSSR,OOE"
-MSSIB,Special,-2,15,"CI,CL,OTH,SEMH,SP,UKN","EO,EYE,FEC,IMS,ISS,ISSR,MAP,MMS,MMSIB,MMSOB,MSS,MSSIB,MSSOB,MSSR,MU,MUOB,NMSS,NMSSR,OOE"
-MSSOB,Special,-2,15,"CI,CL,OTH,SEMH,SP,UKN","EO,EYE,FEC,IMS,ISS,ISSR,MAP,MMS,MMSIB,MMSOB,MSS,MSSIB,MSSOB,MSSR,MU,MUOB,NMSS,NMSSR,OOE"
-MSSOP,Special,-2,15,"CI,CL,OTH,SEMH,SP,UKN",MSSOP
-MSSR,Special,5,15,"CI,CL,OTH,SEMH,SP,UKN","EO,EYE,FEC,IMS,ISS,ISSR,MAP,MMS,MMSIB,MMSOB,MSS,MSSIB,MSSOB,MSSR,MU,MUOB,NMSS,NMSSR,OOE"
-MU,Other,-2,14,"CI,CL,OTH,SEMH,SP,UKN","EO,EYE,FEC,IMS,ISS,ISSR,MAP,MMS,MMSIB,MMSOB,MSS,MSSIB,MSSOB,MSSR,MU,MUOB,NMSS,NMSSR,OOE"
-MUOB,Other,-2,14,"CI,CL,OTH,SEMH,SP,UKN","EO,EYE,FEC,IMS,ISS,ISSR,MAP,MMS,MMSIB,MMSOB,MSS,MSSIB,MSSOB,MSSR,MU,MUOB,NMSS,NMSSR,OOE"
-NMSS,Special,-2,15,"CI,CL,OTH,SEMH,SP,UKN","EO,EYE,FEC,IMS,ISS,ISSR,MAP,MMS,MMSIB,MMSOB,MSS,MSSIB,MSSOB,MSSR,MU,MUOB,NMSS,NMSSR,OOE"
-NMSSR,Special,-2,15,"CI,CL,OTH,SEMH,SP,UKN","EO,EYE,FEC,IMS,ISS,ISSR,MAP,MMS,MMSIB,MMSOB,MSS,MSSIB,MSSOB,MSSR,MU,MUOB,NMSS,NMSSR,OOE"
-OOE,Other,-4,21,"CI,CL,OTH,SEMH,SP,UKN","EO,EYE,FEC,IMS,ISS,ISSR,MAP,MMS,MMSIB,MMSOB,MSS,MSSIB,MSSOB,MSSR,MU,MUOB,NMSS,NMSSR,OOE"
\ No newline at end of file
+setting,setting-group,min-academic-year,max-academic-year,needs,setting->setting
+A,group1,-3,15,"T,U,V,W,X,Y","A,B,C,D,E,F,G,H,I,J,K,L,N,O,P,Q,R,S"

```

Comment:
This looks like all needs and settings transitions are possible, but where did your AYs come from? Does it match the "historic" data?

Reply:
this is just the mins and maxes from each setting in the new data